virttest/utils_libvirt/libvirt_disk.py

```diff
@@ -40,3 +44,70 @@ def create_disk(disk_type, path=None, size="500M", disk_format="raw", extra='',
     else:
         return libvirt.create_local_disk(disk_type, path=path, size=size,
                                          disk_format=disk_format, extra=extra)
+
+
+def create_custom_metadata_disk(disk_path, disk_format,
+                                disk_device, device_target, device_bus, max_size):
+    """
+    Create another disk for a given path,customize driver metadata attribute
+
+    :param disk_path: the path of disk
+    :param disk_format: the format to disk image
+    :param disk_device: the disk device type
+    :param device_target: the target of disk
+    :param device_bus: device bus
+    :param max_size: metadata_cache max size
+    :return: disk object if created successfully
+    """
+    custom_disk = Disk(type_name='file')
+    custom_disk.device = disk_device
+    source_dict = {}
+    if disk_path:
+        source_dict.update({'file': disk_path})
+    custom_disk.source = custom_disk.new_disk_source(
+        **{"attrs": source_dict})
+    target_dict = {"dev": device_target, "bus": device_bus}
+    custom_disk.target = target_dict
+    driver_dict = {"name": "qemu", 'type': disk_format}
+    # Create drivermetadata object
+    new_one_drivermetadata = custom_disk.new_drivermetadata(**{"attrs": driver_dict})
+    metadata_cache_dict = {"max_size": max_size, "max_size_unit": "bytes"}
+    # Attach metadatacache into drivermetadata object
+    new_one_drivermetadata.metadata_cache = custom_disk.DriverMetadata().new_metadatacache(**metadata_cache_dict)
+    custom_disk.drivermetadata = new_one_drivermetadata
+    logging.debug("disk xml in create_custom_metadata_disk: %s\n", custom_disk)
+    return custom_disk
+
+
+def get_images_with_xattr(vm):
+    """
+    Get the image path(s) which still having xattr left of a vm
+
+    :param vm: The vm to be checked
+    :return: The image path list which having xattr left
+    """
+    disks = vm.get_disk_devices()
+    dirty_images = []
+    check_cmd = "getfattr -m trusted.libvirt.security -d %s"
+    for disk in disks:
+        disk_path = disks[disk]['source']
+        getfattr_result = process.run(check_cmd % disk_path, shell=True)
+        if "selinux" in getfattr_result.stdout_text:
+            dirty_images.append(disk_path)
+            logging.debug("Image '%s' having xattr left: %s",
+                          disk_path, getfattr_result.stdout)
+    return dirty_images
+
+
+def clean_images_with_xattr(dirty_images):
+    """
+    Clean the images' xattr
+
+    :param dirty_images: The image list to be cleaned
+    """
+    clean_cmd = ("for i in $(getfattr -m trusted.libvirt.security -d %s |"
```

Comment:
It is not good to use bash script here. I suggest we translate it to python codes.

Reply:
remove it from this PR